### ISSUE TITLE
Server builds don't work in FF (or Safari?)

### ISSUE THREAD

**Issue by markhalliwell** [unicorn-fail/dreditor#148]
This appears to just be a server issue (https://dreditor.org/development/build) as I can locally build both FF and Safari and they work just fine.

I did a diff of a local and site (server) extracted builds and the only thing I could find was:

```
$ diff -r local.dreditor.xpi/ site.dreditor.xpi/
Only in local.dreditor.xpi/: .DS_Store
Only in local.dreditor.xpi/: defaults
diff -r local.dreditor.xpi/install.rdf site.dreditor.xpi/install.rdf
26a27
>
diff -r local.dreditor.xpi/resources/dreditor/data/dreditor.js site.dreditor.xpi/resources/dreditor/data/dreditor.js
12c12
<  * Last build: 2014-05-22 9:16:29 AM CDT

---
>  * Last build: 2014-05-21 7:38:18 PM CDT

```

![img](http://monosnap.com/image/gnkmgJWt5ItrhXdlvfGBdmwyoGoS3e.png)

@sun, @Cottser, maybe you see something I don't?

**Comment by star-szr** [comment]
Currently my work machine is behaving like the dreditor.org server and my laptop at home is working fine (generating the empty defaults/preferences/prefs.js). I will try my older laptop at home to see what it will do. I'm thinking node versions, grunt versions, what other data could we gather?

Broken build:

```
$ node --version
v0.10.28
$ grunt --version
grunt-cli v0.1.13
grunt v0.4.5
```

Working build:

```
$ node --version
v0.10.28
$ grunt --version
grunt-cli v0.1.11
grunt v0.4.2
```

**Comment by markhalliwell** [comment]
Local:

``` bash
$ echo "node: $(node --version)"; echo "npm: $(npm --version)"; grunt --version
node: v0.10.26
npm: 1.4.6
grunt-cli v0.1.11
grunt v0.4.4
```

dreditor.org:

``` bash
$ echo "node: $(node --version)"; echo "npm: $(npm --version)"; grunt --version
node: v0.10.28
npm: 1.4.9
grunt-cli v0.1.9
grunt v0.4.5
```

**Comment by markhalliwell** [comment]
Alrighty, I found the issue (introduced in https://github.com/dreditor/dreditor/commit/a8fcb4120e0209dfd6c85db68f524a93dc7c8694#diff-35b4a816e0441e6a375cd925af50752cL249):
We were chasing https://github.com/mozilla/addon-sdk/tree/master

So when `grunt install` installed the `mozilla-addon-sdk` files in the cloned repo's `./tmp/` it was always using their HEAD, which obviously contains some bugs. Instead I've tested this with the last release on https://github.com/mozilla/addon-sdk/releases: `1.16` and it fixes it. Will submit a PR shortly.
